### PR TITLE
BE add auth process to all routes and update the related controllers

### PR DESCRIPTION
**Pull request by AdrieneTZ**
- Add package: `cors`
- Fix the issue on the authentication process
- Add authentication process on all routes and update all the related controllers
- All routes have passed the thunder client test

### CODE REVIEW

**Comment by AdrieneTZ** [comment]
After merging this branch, how about having a bubble milk tea :DD?

File: back-end/controllers/record-controller.js [hunk 1]
```diff
@@ -22,104 +22,118 @@ const recordController = {
...
+      // Check if the record is in database
       const theRecord = await prisma.record.findUnique({
         where: {
           id: recordId,
         },
       })
       theRecord ? theRecord : res.json('the record does not exist')
 
+      // Check if the record is one of the current user's records
+      userId === theRecord.userId
```
Comment (davipon) [inline back-end/controllers/record-controller.js:91]: It seems you forget to declare userId in this controller before assiging to it.

**Review by davipon — CHANGES_REQUESTED** [review]
`pnpm-lock.json` should be included in this PR.

Make sure to cd to snowball root folder and stage it.

File: back-end/controllers/record-controller.js [hunk 2]
```diff
@@ -3,13 +3,13 @@ const { Prisma, PrismaClient } = require('@prisma/client')
 const prisma = new PrismaClient({ log: ['query'] })
 
 const recordController = {
-  // create a new record
+  // Create a new record
+  // URL: post /records
   postReocrd: async (req, res, next) => {
     try {
-      // #swagger.tags = ['Expense Record']
+      console.log(req.user)
```
[the remark this answers is not in the document]
Comment (AdrieneTZ) [reply back-end/controllers/record-controller.js]: Thank you for reminding me to remove it.

File: back-end/controllers/record-controller.js (hunk 1 above)
Comment (AdrieneTZ) [reply back-end/controllers/record-controller.js:91]: I have added taking `userId` from `req` on the latest commit.

**Review by davipon — APPROVED** [review]
LGTM

**Pull request merged**